crates/e2e/tests/e2e/partially_fillable_observed_score.rs

```diff
@@ -0,0 +1,197 @@
+    );
+    tx!(
+        solver.account(),
+        onchain.contracts().uniswap_v2_router.add_liquidity(
+            token_a.address(),
+            token_b.address(),
+            to_wei(1000),
+            to_wei(1000),
+            0_u64.into(),
+            0_u64.into(),
+            solver.address(),
+            U256::max_value(),
+        )
+    );
+
+    tx!(
+        trader_a.account(),
+        token_a.approve(onchain.contracts().allowance, to_wei(500))
+    );
+
+    let services = Services::new(onchain.contracts()).await;
+    services.start_autopilot(vec![]);
+    services
+        .start_api(vec![
+            "--allow-placing-partially-fillable-limit-orders=true".to_string()
+        ])
+        .await;
+
+    let order_a = OrderBuilder::default()
+        .with_sell_token(token_a.address())
+        .with_sell_amount(to_wei(100))
+        .with_buy_token(token_b.address())
+        .with_buy_amount(to_wei(50))
+        .with_valid_to(model::time::now_in_epoch_seconds() + 300)
+        .with_kind(OrderKind::Sell)
+        .with_partially_fillable(true)
+        .sign_with(
+            EcdsaSigningScheme::Eip712,
+            &onchain.contracts().domain_separator,
+            SecretKeyRef::from(&SecretKey::from_slice(trader_a.private_key()).unwrap()),
+        )
+        .build()
+        .into_order_creation();
+    services.create_order(&order_a).await.unwrap();
+
+    tracing::info!("Waiting for order to show up in auction.");
+    let has_order = || async { services.get_auction().await.auction.orders.len() == 1 };
+    wait_for_condition(TIMEOUT, has_order).await.unwrap();
+
+    let auction = services.get_auction().await.auction;
+    let order = auction.orders.into_iter().next().unwrap();
+    assert!(order.data.partially_fillable);
+    assert!(matches!(order.metadata.class, OrderClass::Limit(_)));
+    assert_eq!(order.metadata.full_fee_amount, 0.into());
+    assert_eq!(order.metadata.solver_fee, 0.into());
+    assert_eq!(auction.rewards.get(&order.metadata.uid), None);
+
+    tracing::info!("Waiting for trade.");
+    services.start_old_driver(solver.private_key(), vec!["--solvers=Baseline".to_owned()]);
+    let trade_happened =
+        || async { token_b.balance_of(trader_a.address()).call().await.unwrap() != 0.into() };
+    wait_for_condition(TIMEOUT, trade_happened).await.unwrap();
+
+    let get_block = || async {
+        web3.eth()
+            .block(BlockId::Number(ethcontract::BlockNumber::Latest))
+            .await
+            .unwrap()
+            .unwrap()
+    };
+    let settlement_tx_1 = get_block().await.transactions.pop().unwrap();
+
+    // Expecting a partial fill because order sells 100 but user only has 50.
+    let sell_balance = token_a
+        .balance_of(trader_a.address())
+        .call()
+        .await
+        .unwrap()
+        .to_f64_lossy();
+    // Depending on how the solver works might not have sold all balance.
+    assert!((0e18..=1e18).contains(&sell_balance));
```

Comment:
In this test you are assuming the exact balance that was sold (50). This happens in the asserts below. To be consistent with that the asserts here (that were copied from the other test) should be changed. Maybe remove them.

Reply:
I implemented this.